src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -101,13 +271,60 @@ def algorithm(self, parameters=None):
             with CodeTimer("Rucio block resolution", logger=logging):
                 self.findRucioBlocks(uniqueActivePU, msPileupList)
 
-            # TODO in future tickets: find the json files in the spec/sandbox area and tweak them
-
+            # loop over existing pileup names and
+            # find the json files in the spec/sandbox area and tweak them
+            adjustJSONSpec(msPileupList)
         except Exception as ex:
             msg = f"Caught unexpected exception in WorkflowUpdater. Details:\n{str(ex)}"
             logging.exception(msg)
             raise WorkflowUpdaterException(msg) from None
 
+    def adjustJSONSpec(self, msPileupList):
+        """
+        Main logic of the algorithm:
+        - for every pileup record find out location of tarball
+        - get configuration files within tarball
+        - extract block information
+        - replace pileupconf.json files within tarball
+        :param msPileupList: input list of pileup records
+        :return: nothing, it perform checks and adjust in place JSON spec file(s)
+        """
+        # loop over existing pileup names and
+        # find the json files in the spec/sandbox area and tweak them
+        for rec in msPileupList:
+            # find out location of pileup tarball
+            files = findFiles(self.idir, rec['pileupName'])
```

Comment:
The reason for scan is unknown structure of the document. I do not know it a-priory, I also do not know if it will change (it might). I also don't think it cost anything with respect of the entire algorithm code flow, i.e. rucio call can be more expensive that scan of tar ball. Said that, I do not want to overcomplicate code with assumption about unknown data structure.

Reply:
No, this is wrong! Here you are scanning ALL workflows in the agent for each pileup retrieved from MSPileup. This is going to be a huge I/O load on the node and very very inefficient (Rucio calls is nothing compared to this, even if everything is sequential).

Please analyze the code we have so far such that we can:
* scan **the right** sandbox/tarball
* for **the right** pileup name
 
scan a given sandbox (or workflow tarball), instead of the whole directory with ALL tarballs/sandboxes of the agent.

IF needed, I would invite you to redeploy your test/testbed agent, apply this patch over there and verify which information exactly is already collected in this new component. Or whatever way it works the best for you.

Reply:
If I understood your suggestion correctly I applied it in this commit: https://github.com/dmwm/WMCore/pull/11884/commits/ace826a9f59d8b7fd42c5eee19938ce65b82b859 Please have a look. It reduces the number the number of adjustements ot ones defined in `puWflows`.

Reply:
This is resolved in new codebase.